File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/BulletHitWallEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BulletState;

/** Event occurring when a bullet has hit a wall */
@SuppressWarnings("unused")
public final class BulletHitWallEvent extends Event {

  /** Bullet that has hit a wall */
  private final BulletState bullet;

  public BulletHitWallEvent(int turnNumber, BulletState bullet) {
    super(turnNumber);
    this.bullet = bullet;
  }

  /** Returns the bullet that has hit a wall */
  public BulletState getBullet() {
    return bullet;
  }
}

File: bot-api/dotnet/api/src/events/BulletHitWallEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when a bullet has hit a wall.
/// </summary>
public sealed class BulletHitWallEvent : BotEvent
{
    /// <summary>Bullet that has hit a wall.</summary>
    public BulletState Bullet { get; }

    /// <summary>
    /// Initializes a new instance of the BulletHitWallEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that has hit a wall.</param>
    [JsonConstructor]
    public BulletHitWallEvent(int turnNumber, BulletState bullet) : base(turnNumber) => Bullet = bullet;
}